File: packages/typia/src/transformers/features/misc/MiscAssertPruneTransformer.ts
import { MiscAssertPruneProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace MiscAssertPruneTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "misc.assertPrune",
      write: MiscAssertPruneProgrammer.write,
    });
}

File: packages/transform/src/features/misc/MiscAssertPruneTransformer.go
package misc

import (
	coremisc "github.com/samchon/typia/packages/core/src/programmers/misc"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var MiscAssertPruneTransformer = miscAssertPruneTransformerNamespace{}

type miscAssertPruneTransformerNamespace struct{}

func (miscAssertPruneTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return coremisc.MiscPruneProgrammer.Write(call.TypeText)
}
